build/pack.ps1

```diff
@@ -10,7 +10,7 @@ Write-Host "##[info]Copy Native simulator xplat binaries"
 pushd ../src/Simulation/Native
 If (-not (Test-Path 'osx')) { mkdir 'osx' }
 If (-not (Test-Path 'linux')) { mkdir 'linux' }
-$DROP = "$Env:DROP_NATIVE/src/Simulation/Native/build"
+$DROP = "$Env:DROP_NATIVE/src/Simulation/Native/build/drop"
```

Comment:
To unify consumption of the native simulator library across platforms, I added 'install' command to the cmake files so the dll and tests can be "installed" into a known folder.

Reply:
actually, I think the change here is ok, as it seems we don't really copy the dlls here anymore, they are copied during bootstrap (so this might not be necessary at all). I'm a little confused why they are getting copied, though, since bootstrap.ps1 the last thing it does is call intsll-prereqs on QIR, and QIR doesn't have this code; any ideas of where this is coming from?
https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=19385&view=logs&j=663ee715-dcc5-5b8a-8432-471131753a80&t=ebac3767-3683-5916-990d-44e0e78421f8&l=1492

Reply:
I am trying to solve two problems here:
1. depending on the context (ci/e2e/local) we might build either DEBUG or RELEASE build of native simulator but the consumers wouldn't know which one to pick up
2. the target folder of cmake build is created differently on Windows vs Linux and the install into "drop" folder helps unifying the handling across platforms.

My grasp of who copies what and when during the pipeline is rather shaky. The build kept failing and I just searched through the repo for mentions of "libMicrosoft.Quantum.Simulator.Runtime.so" and re-pointed them to pick up from the unified drop location.